app/controllers/submissions_controller.rb

```diff
@@ -472,11 +472,26 @@ def view
     @curSubmissionIndex = @latestSubmissions.index do |submission|
       submission.course_user_datum.user.email == @submission.course_user_datum.user.email
     end
-    @prevSubmission = @curSubmissionIndex > 0 ? @latestSubmissions[@curSubmissionIndex - 1] : nil
+    @prevSubmission = if @curSubmissionIndex > 0
+                        @latestSubmissions[@curSubmissionIndex - 1]
+                      end
     @nextSubmission = if @curSubmissionIndex < (@latestSubmissions.size - 1)
                         @latestSubmissions[@curSubmissionIndex + 1]
                       end
 
+    @userVersions = @assessment.submissions
+                               .where(course_user_datum_id: @submission.course_user_datum_id)
+                               .order("version DESC")
+    @curVersionIndex = @userVersions.index do |submission|
+      submission.version == @submission.version
+    end
+    @prevVersion = if @curVersionIndex < (@userVersions.size - 1)
```

Comment:
this two variables `prevVersion` and `nextVersion` are not used in this PR

Reply:
Just realised that too -- will remove!